tests/test_multiresolution.py

```diff
@@ -86,7 +86,7 @@ def test_surveys(self):
                     coverage="union",
                 )
                 interp_scar = obs_lr.render(data_hr[None, :, :])
-                assert SDR(interp_scar, data_lr) > 10
+                # assert SDR(interp_scar, data_lr) > 10
```

Comment:
Actually, this means that there is a problem. Most likely misalignment causing shifts.

Reply:
Actually, this meansthat there is a problem. Most likely observations are not aligned and there is a shift.

Reply:
I agree, but I think the problem is not in code but in the test data. It uses a custom WCS stored in the file, which I think you told me is chosen to work well with `Frame.from_observations`. In this branch, I have replaced the offset of the mode frame by a WCS with a shifted reference point. This is the only way of making a consistent model frame with more than two observations. I'm pretty sure I got it right and the multi-resolution tutorial seems to agree with me, but this test fails. Can you confirm how the WCS for `TestLowResObservation` have been constructed?

Reply:
It uses my simulations tool, `mr_tools` which is part of my [multi-resolution package
](https://github.com/herjy/Multi_resolution_comparison). I'd have to check exactly what you have done in your modifications to wcs. I'll also "look at the damn" by checking the residuals.

Reply:
Yes, I have. I am confused by what I see and I don't like it. Here's the test data (last image, which throws the assert): with `coverage="union"`:
![download](https://user-images.githubusercontent.com/1463403/104381967-5684cb80-54fb-11eb-90dc-a616c0844354.png)
First panel: `data_lr` directly from the test set, second panel the rendered image from scarlet, third the difference. This is a clear dipole. It's present in some of the other test images as well, but this one is the largest residual, and the only one with `SDR < 10`. This indicates tow things to me: SDR 10 is not sharp enough, and this is likely related to the parity discussion from below!.

Now with `coverage="intersection"`:
![download-1](https://user-images.githubusercontent.com/1463403/104381990-600e3380-54fb-11eb-85b6-8a9c3c55e764.png)
What confuses me is that both intersection and coverage yield identical cutouts, which cannot generally be the case. What confuses me even more is that despite the same shape, the residuals are different. Something is very weird here. I will keep the line commented out, but this need to get fixed!

Reply:
This sounds like headaches for me. I'll address this.

Reply:
Another wrinkle. The test data have some WCSs with non-integer CRPIX. Was that intended? It's not illegal but uncommon to shift the reference.

Reply:
So the problem was caused by an rounding error and integer conversion in `from_observations`. It's test-specific in that the tests takes an image in HR with its own WCS and seeks to render it to LR to compare it to a LR image with its WCS. But `render` goes from *model* frame to LR frame, not from HR frame to LR frame. If the model frame isn't identical to the HR frame, this will go wrong. The test triggered it, and it's now fixed, but the problem is somewhat artifical.